src/generator/AutoRest.Ruby/Templates/ServiceClientTemplate.cshtml

```diff
@@ -79,6 +79,62 @@ module @Settings.Namespace
     end
 
     @EmptyLine
+    #
+    # Makes a request and returns the body of the response
+    # @@param method [Symbol] with any of the following values :get, :put, :post, :patch, :delete   
+    # @@param path [String] the path, relative to {api_endpoint}
+    # @@param options [Hash{String=>String}] specifying any request options like :body
+    # @@return [Hash{String=>String}] containing the body of the response
+    #
+    def make_request(method, path, options = {})
+      result = make_request_async(method, path, options).value!
+      result.response.body.to_s.empty? ? nil : JSON.load(result.response.body)
+    end
+    @EmptyLine
+
+    #
+    # Makes a request asynchronously
+    # @@param method [Symbol] with any of the following values :get, :put, :post, :patch, :delete
+    # @@param path [String] the path, relative to {api_endpoint}
+    # @@param options [Hash{String=>String}] specifying any request options like :body
+    # @@return [Concurrent::Promise] Promise object which holds the HTTP response.
+    #
+    def make_request_async(method, path, options = {})
+      fail ArgumentError, 'method is nil' if method.nil?
+      fail ArgumentError, 'path is nil' if path.nil?
+
+      @EmptyLine
+      request_headers= REQUEST_HEADERS
+      request_headers.merge!({'accept-language' => @@accept_language}) unless @@accept_language.nil?
+
+      @EmptyLine
+      request_url= options[:api_endpoint] || @@base_url
+      
+      @EmptyLine
+      options=options.merge({
```

Comment:
I need to split the first line you called out above, so that I get the initial default headers (even if the condition (`unless @accept_language.nil?`) evaluates to false. Rest looks good, so I'm updating with that.